packages/compass/src/main/application.ts

```diff
@@ -230,6 +230,9 @@ class CompassApplication {
         debug('Did not agree to license, quitting app.');
         app.quit();
       },
+      'compass:check-secret-storage-is-available': function () {
+        return safeStorage.isEncryptionAvailable();
```

Comment:
Looking at https://github.com/electron/electron/issues/33640 it sounds like if we haven't had the `whenReady` event this can return false, at least on windows. Since we're using it from the renderer I don't think this will be an issue, although we may want to add a promise to return this value when it's ready if it's not yet ready.
At the least I think it's worth calling out in a comment, although I do feel it could make it a bit prone to a future bug if we use it differently, so I think making sure it's ready first is good.
```suggestion
        // This will return `false` even if a credential store is available on windows if this is called
        // before electron's `whenReady`. https://github.com/electron/electron/issues/33640
        return safeStorage.isEncryptionAvailable();
```

Reply:
It is safe to call `isEncryptionAvailable` here since the app is already ready and rendered, but I have left a comment for visibility. I also added `await app.whenReady()` before logging `SafeStorage initialized` so we have a proper value on Windows.